docs/troubleshoot/app-framework/appserver-error-codes.md

```diff
@@ -2388,283 +2486,332 @@ The following error message codes may appear on the app-server log. Use the foll
+  Skipping authentication plugin _pluginId_ because it's not HA compatible
 
   **Reason:**
 
-  TODO
+  The server is setup for running in high availability (HA) mode which requires that plugins that have state, in particular authentication plugins, must be HA-compatible or else errors will occur. Therefore, the server skips over loading of this plugin nbecause its pluginDefinition.json did not state it was HA compatible.
 
   **Action:**
 
-  TODO
+  Either the plugin must be updated to support and state its support for HA, or it must be removed, or HA mode disabled. To make a plugin support HA, the conformance program should be reviewed. When HA mode is supported, the plugin can be marked as compatible by setting capabilities.haCompatible=true within its initialization.
 
 
 
-### ZWED0178W":"Skipping authentication plugin %s because it's not HA compatible
+### ZWED0179W
+
+  Unable to retrieve the list of certificate authorities from the keyring=_keyringName_ owner=_username_ Error: _error_
 
   **Reason:**
 
-  TODO
+  The server could not automatically determine the certificate authorities (CA) from the z/OS keyring listed. This may cause the server to be unable to verify certificate chains from other servers or clients causing other errors later.
 
   **Action:**
 
-  TODO
+  Review the error to resolve it and contact support if needed. It's also possible as a workaround to explicitly state the CAs within the keyring that you would like to load, rather than relying upon the server's attempt to automatically find all CAs within the keyring.
 
 
 
-### ZWED0179W":"Unable to retrieve the list of certificate authorities from the keyring=%s owner=%s Error: %s
+### ZWED0001E
+
+  Error: _error_
 
   **Reason:**
 
-  TODO
+  The server is running in cluster mode and the cluster manager has encountered an unexpected error.
 
   **Action:**
 
-  TODO
+  Review the error to resolve it, and contact support if needed.
 
 
 
-### ZWED0001E":"Error: %s
+### ZWED0002E
+
+  Could not stop language manager for types=_languageName_
 
   **Reason:**
 
-  TODO
+  A plugin had a service that needed a language manager to run. During shutdown, the language manager could not be stopped.
 
   **Action:**
 
-  TODO
+  The language manager may continue to run after the app-server shuts down. Review the logs to determine the location of the language manager and try to stop the manager manually.
 
 
 
-### ZWED0002E":"Could not stop language manager for types=%s
+### ZWED0003E
+
+  Loopback configuration not valid, _loobackConfiguration_
+  Loopback calls will fail!
 
   **Reason:**
 
-  TODO
+  The loopback configuration that the server uses to contact itself over an internal network was missing a value for the network port, therefore no requests over the loopback address will be possible.
 
   **Action:**
 
-  TODO
+  Review the configuration of components.app-server.node.port to see if it has a value and set one to fix the issue.
```

Comment:
Btw, it's probably good to code format the components.app-server.node.port

I think it's the ` character in markdown but I could be wrong

Reply:
it is